src/Php/PhpVersionFactoryFactory.php

```diff
@@ -29,17 +28,15 @@ public function create(): PhpVersionFactory
 		$composerPhpVersion = null;
 		if (count($this->composerAutoloaderProjectPaths) > 0) {
 			$composerJsonPath = end($this->composerAutoloaderProjectPaths) . '/composer.json';
-			if (is_file($composerJsonPath)) {
-				try {
-					$composerJsonContents = FileReader::read($composerJsonPath);
-					$composer = Json::decode($composerJsonContents, Json::FORCE_ARRAY);
-					$platformVersion = $composer['config']['platform']['php'] ?? null;
-					if (is_string($platformVersion)) {
-						$composerPhpVersion = $platformVersion;
-					}
-				} catch (CouldNotReadFileException | JsonException) {
-					// pass
+			try {
+				$composerJsonContents = FileReader::read($composerJsonPath);
+				$composer = Json::decode($composerJsonContents, Json::FORCE_ARRAY);
+				$platformVersion = $composer['config']['platform']['php'] ?? null;
+				if (is_string($platformVersion)) {
+					$composerPhpVersion = $platformVersion;
 				}
+			} catch (CouldNotReadFileException | JsonException) {
```

Comment:
we already handle the non-readable case via the try/catch, no need for the additional IO before calling into `FileReader::read`